### ISSUE TITLE
HDFView 3.1.3 with HDF5 1.10.7 cannot load JLD2

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaIO/JLD2.jl#381]
The readme claims JLD2 files should be able to be loaded by viewers using the HDF5 1.10 library, but it doesn't seem to work.

![image](https://user-images.githubusercontent.com/1438610/154302904-e12fd457-5dda-41b0-8afe-5c6371fc8f5c.png)
![image](https://user-images.githubusercontent.com/1438610/154303081-86c67289-ca7a-428d-b8d5-361cae980def.png)

Maybe I'm doing something wrong? I'm not that familiar with the HDF2 format.

**Comment by JonasIsensee** [comment]
Hi @BioTurboNick,
thanks. I looked into this a bit and conclude that this is not a problem of JLD2
but rather a limitation of HDFView.

What happens is the following:
Your toplevel object `results` is an array with a single element which is itself an array.
Arrays are not fixed-size and therefore stored as references.
Sadly HDFView does not seem to be able to appropriately recursively de-reference that.

`h5ls` seems to at least be able to recursively go through the arrays but stops when it encounter the custom compound types (see the `, ,` indicating a not-loaded field)

```
$ h5ls -d Well\ B\ 002.bin.txt\ raw\ data.jld2
_types                   Group
results                  Dataset {1}
    Data:
         DATASET "Well B 002.bin.txt raw data.jld2"
            DATA {
               DATASET "Well B 002.bin.txt raw data.jld2"
                  DATA {
                     {"488", , 20415, 97.0159396258631, 210.429338505913, 1377, 14.1935439197964, 10.1938997821351, 
                     1951.5891952287, 0.312417340191036, 0.94979841835023, 7657.95190296501, 21.6711878507257, , , , 
                     , , , , }
                  }
               DATASET "Well B 002.bin.txt raw data.jld2"
                  DATA {
                     {"561", , 5485, 97.0159396258631, 56.5371012346282, 305, 3.14381328651989, 14.144262295082, 
                     1769.2656479088, 0.213491340018232, 0.969715760367947, 11433.0327858515, 199.92960616373, , , , 
                     , , , , }
                  }
               DATASET "Well B 002.bin.txt raw data.jld2"
                  DATA {
                     {"647", , 4034, 97.0159396258631, 41.5807960584303, 188, 1.93782589464177, 15.8244680851064, 
                     2004.1744421535, 0.26251859196827, 0.95928693808043, 8715.92598189458, 229.628009028727, , , , 
                     , , , , }
                  }
            }
```


Long story short:
If you want to look at your files using hdf5 tools, make sure to not nest mutable objects by e.g. unrolling your data into different datasets.

**Comment by BioTurboNick** [comment]
Okay great, thank you!
